### PR TITLE
Import: prevent recursion blowup in redux + reintroduce import e2e test

### LINKED ISSUE #52976: Dashboard import results in redux recursion error
**What happened**:

image

**What you expected to happen**:

it to import successfully

**How to reproduce it (as minimally and precisely as possible)**:

attempt to import https://raw.githubusercontent.com/grafana/grafana-javascript-agent/main/dashboards/frontend-application.json

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- reenables the import e2e test

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/52976

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/selectors/navModel.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ function getSectionRoot(node: NavModelItem): NavModelItem {
   if (root.children) {
     root.children = root.children.map((item) => {
       if (item.id === node.id) {
-        return { ...node, active: true };
+        return { ...item, active: true };
```
Comment (torkelo) [inline public/app/core/selectors/navModel.ts:46]: This should break a unit test, as item is missing the updated children prop (with active set)

Comment (torkelo) [reply public/app/core/selectors/navModel.ts:46]: If I remember correctly

Comment (ashharrison90) [reply public/app/core/selectors/navModel.ts:46]: yeah - this is basically a draft for now. i want to get the e2e test reenabled first 😄

**Comment by ashharrison90** [comment]
@torkelo have reenabled the e2e test, now looking into the implementation. can i clarify what the intention of the code is? it looks like it's a combination of setting `active: true` on the node (and any parents?) that's passed in and returning the top-most `parentItem`?

**Comment by ashharrison90** [comment]
as far as i can tell, the second level children not having the active prop set will only affect behaviour when `topnav` is set. with that in mind, i think a bug in `topnav` is better than a pretty serious bug in main. so i'm gonna disable the unit test for now to deliver the "fix" for main, then we have more time to iron out the logic. wdyt?

**Review by torkelo — APPROVED** [review]
Sure

**Pull request merged**